pcawg_metadata_parser/pc_report-transfer_summary.py

```diff
@@ -0,0 +1,221 @@
+                report[project]['both_not_transferred']['count'] -= 1 
+                report[project]['both_not_transferred']['donors'].discard(donor_unique_id)   
+
+    report_dir = init_report_dir(metadata_dir, report_name, repo)
+    
+    summary_table = []
+    for p in report.keys():
+        summary = OrderedDict()
+        summary['project'] = p
+        summary['timestamp'] = timestamp
+        for ctype in count_types:
+            summary[ctype] = report.get(p).get(ctype).get('count') if report.get(p).get(ctype) else 0
+            donors = report.get(p).get(ctype).get('donors') if report.get(p).get(ctype) else []
+
+            if donors:
+                with open(report_dir + '/' + p + '.' + ctype + '.donors.txt', 'w') as o:
+                    o.write('# ' + ctype + '\n')
+                    o.write('# dcc_project_code' + '\t' + 'submitter_donor_id' + '\n')
+                    for d in donors:
+                        # TODO: query ES to get JSON then retrieve BAM info: aligned/unaligned, gnos, bam file name etc
+                        o.write(d.replace('::', '\t') + '\n')
+
+        summary_table.append(summary)
+
+    with open(report_dir + '/donor.json', 'w') as o:
+        o.write(json.dumps(summary_table))
+
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv
+    else:
+        sys.argv.extend(argv)
+
+    parser = ArgumentParser(description="PCAWG Report Generator Using ES Backend",
+             formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument("-m", "--metadata_dir", dest="metadata_dir",
+             help="Directory containing metadata manifest files", required=True)
+    parser.add_argument("-r", "--gnos_repo", dest="repo",
+             help="Specify which GNOS repo to process, process all repos if none specified", required=False)
+
+    args = parser.parse_args()
+    metadata_dir = args.metadata_dir  # this dir contains gnos manifest files, will also host all reports
+    repo = args.repo
+
+    if not os.path.isdir(metadata_dir):  # TODO: should add more directory name check to make sure it's right
+        sys.exit('Error: specified metadata directory does not exist!')
+
+    timestamp = str.split(metadata_dir, '/')[-1]
+    es_index = 'p_' + ('' if not repo else repo+'_') + re.sub(r'\D', '', timestamp).replace('20','',1)
+    es_type = "donor"
+    es_host = 'localhost:9200'
+
+    es = Elasticsearch([es_host])
+
+    report_name_base = re.sub(r'^pc_report-', '', os.path.basename(__file__))
+    report_name_base = re.sub(r'\.py$', '', report_name_base)
+
+
+    # read and parse git for the gnos_ids and fnames which are completed for data transfer
+    for transfer_target in ['s3', 'ceph']:
+        if transfer_target == 's3':
+            git_fnames = '../s3-transfer-operations/s3-transfer-jobs*/completed-jobs/*.json'
+        else:
+            git_fnames = '../ceph_transfer_ops/ceph-transfer-jobs/completed-jobs/*.json'
```

Comment:
Very good idea.

-Linda






On Oct 6, 2015, at 10:46 AM, Junjun Zhang <notifications@github.com<mailto:notifications@github.com>> wrote:


In pcawg_metadata_parser/pc_report-transfer_summary.py<https://github.com/ICGC-TCGA-PanCancer/pcawg-central-index/pull/37#discussion_r41273475>:

> +    es_index = 'p_' + ('' if not repo else repo+'_') + re.sub(r'\D', '', timestamp).replace('20','',1)
> +    es_type = "donor"
> +    es_host = 'localhost:9200'
> +
> +    es = Elasticsearch([es_host])
> +
> +    report_name_base = re.sub(r'^pc_report-', '', os.path.basename(__file__))
> +    report_name_base = re.sub(r'\.py$', '', report_name_base)
> +
> +
> +    # read and parse git for the gnos_ids and fnames which are completed for data transfer
> +    for transfer_target in ['s3', 'ceph']:
> +        if transfer_target == 's3':
> +            git_fnames = '../s3-transfer-operations/s3-transfer-jobs*/completed-jobs/*.json'
> +        else:
> +            git_fnames = '../ceph_transfer_ops/ceph-transfer-jobs/completed-jobs/*.json'


We can make the folder name accept * as well: ceph-transfer-jobs*. Just in case, this is needed in the future, the code already supports.

—
Reply to this email directly or view it on GitHub<https://github.com/ICGC-TCGA-PanCancer/pcawg-central-index/pull/37/files#r41273475>.